zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/Commands.java

```diff
@@ -743,7 +744,14 @@ public CommandResponse run(ZooKeeperServer zkServer, Map<String, String> kwargs)
                                 && voter.addr.equals(peer.getQuorumAddress())
                                 && voter.electionAddr.equals(peer.getElectionAddress())
                 );
+                response.put("myid", zkServer.getConf().getServerId());
+                response.put("is_leader", zkServer instanceof LeaderZooKeeperServer);
+                response.put("endpoint", peer.getElectionAddress().toString().split(":")[0]+ ":" + peer.getClientPort());
```

Comment:
I would council you to omit or refactor "endpoint". The client address can be configured separately from the election address so this may be inaccurate and if the server is using the secure client port only then the address/port combination here also will not respond.

Reply:
Good catch